relayer/chains/evm/provider.go

```diff
@@ -202,126 +199,33 @@ func (cp *EVMChainProvider) RelayPacket(ctx context.Context, packet *bandtypes.P
-
-				// update db as success
-				if err := cp.saveTransaction(ctx, freeSigner.GetAddress(), balance, packet, result); err != nil {
-					log.Error("saveTransaction error", zap.Error(err), zap.Int("retry_count", retryCount))
-				}
-
-				log.Info(
-					"Packet is successfully relayed",
-					zap.String("tx_hash", txHash),
-					zap.Int("retry_count", retryCount),
-				)
-				return nil
-			case chainstypes.TX_STATUS_FAILED:
-				// track transaction processing time (ms)
-				relayermetrics.ObserveTxProcessTime(
-					packet.TunnelID,
-					cp.ChainName,
-					chainstypes.TX_STATUS_FAILED.String(),
-					time.Since(createdAt).Milliseconds(),
-				)
-
-				// track gas used for the relayed transaction
-				relayermetrics.ObserveGasUsed(packet.TunnelID, cp.ChainName, chainstypes.TX_STATUS_FAILED.String(), gasUsed.Decimal.InexactFloat64())
-
-				if err := cp.saveTransaction(ctx, freeSigner.GetAddress(), balance, packet, result); err != nil {
-					log.Error("saveTransaction error", zap.Error(err), zap.Int("retry_count", retryCount))
-				}
-
-				log.Debug(
-					"Transaction failed during relay attempt",
-					zap.Error(err),
-					zap.String("tx_hash", txHash),
-					zap.Int("retry_count", retryCount),
-				)
-				break checkTxLogic
-			case chainstypes.TX_STATUS_PENDING:
-				// update db as pending
-				if !savedOnce {
-					if err := cp.saveTransaction(ctx, freeSigner.GetAddress(), balance, packet, result); err != nil {
-						log.Error("saveTransaction error", zap.Error(err), zap.Int("retry_count", retryCount))
-					} else {
-						savedOnce = true
-					}
-				}
-
-				log.Debug(
-					"Waiting for tx to be mined",
-					zap.Error(err),
-					zap.String("tx_hash", txHash),
-					zap.Int("retry_count", retryCount),
-				)
-
-				time.Sleep(cp.Config.CheckingTxInterval)
-			}
+		if err := cp.saveUnconfirmedTransaction(txHash, types.TX_STATUS_PENDING, packet); err != nil {
+			log.Error("saveTransaction error", zap.Error(err), zap.Int("retry_count", retryCount))
 		}
 
-		// increment the transactions count metric
-		relayermetrics.IncTxsCount(packet.TunnelID, cp.ChainName, txStatus.String())
+		txResult, err := cp.waitForTx(ctx, txHash, createdAt, log, retryCount)
+		if err != nil {
+			log.Error(
+				"Failed to relaying a packet with status and error",
+				zap.Error(err),
+				zap.String("status", txResult.Status.String()),
+				zap.String("tx_hash", txHash),
+				zap.Int("retry_count", retryCount),
+			)
 
-		if err := cp.saveTransaction(ctx, freeSigner.GetAddress(), balance, packet, NewTxResult(txHash, chainstypes.TX_STATUS_TIMEOUT, decimal.NullDecimal{}, decimal.NullDecimal{}, nil)); err != nil {
-			log.Error("saveTransaction error", zap.Error(err), zap.Int("retry_count", retryCount))
+			// bump gas and retry
+			gasInfo, err = cp.BumpAndBoundGas(ctx, gasInfo, cp.Config.GasMultiplier)
+			if err != nil {
+				log.Error("Cannot bump gas", zap.Error(err), zap.Int("retry_count", retryCount))
+			}
```

Comment:
if status is failed, CheckConfirmedTx doesn't return any error and the gas isn't bumped, right? should we log an error and check the txResult.Status instead